### PR TITLE
Add array option

### PR DESCRIPTION
**Pull request by pohantw**
Support even number of tracks by modifying the connections inside SB.

Original method will result in closed loop when the number of tracks is even and make routing fail.

### CODE REVIEW

File: canal/cyclone.py [hunk 1]
```diff
@@ -1041,10 +1041,17 @@ def get_imran_sb_wires(num_tracks: int) -> List[Tuple[int,
 
         for track in range(num_tracks):
             # f_e1
-            result.append((track, SwitchBoxSide.WEST,
-                           mod(w - track, w), SwitchBoxSide.NORTH))
-            result.append((mod(w - track, w), SwitchBoxSide.NORTH,
-                           track, SwitchBoxSide.WEST))
+            if track != 0 and track != w - 1 or mod(w, 2) != 0:
```
Comment (Kuree) [inline canal/cyclone.py:1044]: Use parenthesis when `and` and `or` are used together to improve readability.

File: canal/cyclone.py [hunk 2]
```diff
@@ -1041,10 +1041,17 @@ def get_imran_sb_wires(num_tracks: int) -> List[Tuple[int,
 
         for track in range(num_tracks):
             # f_e1
-            result.append((track, SwitchBoxSide.WEST,
```
Comment (Kuree) [inline canal/cyclone.py:1044]: I'd prefer you create a new topology function called `ImranEven` or something like that and don't touch this function.

**Review by Kuree — CHANGES_REQUESTED** [review]
You should cherry-pick this patch and apply it on top of the tree. No merge commits, only rebase allowed.

**Comment by pohantw** [comment]
I will create a new branch based on latest master and do the PR again

**Pull request closed without merging**